providers/implementations/ciphers/ciphercommon_block.c

```diff
@@ -120,18 +128,19 @@ int ossl_cipher_unpadblock(unsigned char *buf, size_t *buflen, size_t blocksize)
      * Otherwise it provides a padding oracle.
      */
     pad = buf[blocksize - 1];
-    if (pad == 0 || pad > blocksize) {
-        ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-        return 0;
-    }
-    for (i = 0; i < pad; i++) {
-        if (buf[--len] != pad) {
-            ERR_raise(ERR_LIB_PROV, PROV_R_BAD_DECRYPT);
-            return 0;
-        }
+    gp = ~(constant_time_is_zero_s(pad) | constant_time_lt_s(blocksize, pad));
+
+    for (i = 0; i < blocksize; i++) {
+        i_pad_lt = constant_time_lt(i, pad);
```

Comment:
I must not be getting something.. Is this i_pad_lt line right? (If blocksize was 16 and pad was 5 how is this mapping to say that the last 5 bytes in the buffer are 5 : This looks like it does the first 5 bytes)..

Reply:
(Internal tests may be required for this)